src/features/venue/pages/VenueBody.tsx

```diff
@@ -58,7 +58,7 @@ export const VenueBody: FunctionComponent<Props> = ({ venueId, onScroll }) => {
     <Container
       testID="venue-container"
       scrollEventThrottle={20}
-      scrollIndicatorInsets={{ right: 1 }}
+      scrollIndicatorInsets={scrollIndicatorInsets}
```

Comment:
Qu'est-ce qui fait que ça améliore les perfs de simplement l'extraire ?

Reply:
Tu peux lire [cet article](https://hackernoon.com/optimize-props-references-and-improve-rendering-in-react-native-xm3331hn) et si t'as des questions hésites pas.

Mais sinon en gros quand React cherche quelques composant il doit re-render, il fait une comparaison `===` des props et du state. Là la props `{right: 1}` change à chaque render (nouvel référence à chaque fois, ce qui provoque un re-rendering du composant et des enfants.

Comparaison des types primitifs: 
![image](https://user-images.githubusercontent.com/10118284/130478998-2eafa532-e472-46b6-95d0-0c0412c5c817.png)

Donc si tu passes `props="right"` par exemple c'est ✅.
Si tu passes `props={{flex: 1}}` c'est ❌ => donc faut le définir en dehors comme ça la référence change pas.
⚠️ quand tu passes un string du coup, il faut le passer comme ça `props="right"` et non pas comme ça `props={'right'}` parce que là la référence change.

Et sinon je me suis permis de faire cette optim parce que c'est une optim classique mais en général avant de faire une optimisation, il faut toujours mesurer l'impact (donc avant/après) sinon ça peut être dangereux.

Comme outil tu peux notamment utiliser `why-did-you-render` qu'on [a installé ici](https://github.com/pass-culture/pass-culture-app-native/blob/f7d4077d48c1987c86bc8e05efccb884bd8d266e/src/why-did-you-render.js#L6).

Reply:
Top merci